compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1861,6 +1862,16 @@ impl CheckAttrVisitor<'_> {
         }
     }
 
+    fn check_link_ordinal(&self, attr: &Attribute, _span: Span, target: Target) -> bool {
+        match target {
+            Target::ForeignFn => true,
```

Comment:
`link_ordinal` can also be applied to `ForeignStatic`: #100073

Reply:
Fixed and add testcase for it.